mirgecom/initializers.py

```diff
@@ -840,3 +841,149 @@ def __call__(self, x_vec, eos, *, t=0.0):
 
         return join_conserved(dim=self._dim, mass=mass, energy=energy,
                               momentum=mom, species_mass=specmass)
+
+
+class Discontinuity:
+    r"""Solution initializer for flow with a discontinuity.
+
+    This initializer creates a physics-consistent mixture solution
+    given an initial thermal state (pressure, temperature) and a
+    mixture-compatible EOS.
+
+    The solution varies across a planar interface defined by a tanh fucntion
+    located at x=xloc for density, velocity, and pressure
+
+    .. automethod:: __init__
+    .. automethod:: __call__
+    """
+
+    def __init__(
+            self, *, dim=3, xdir=0, x0=0, nspecies=0,
+            tl=300.0, tr=600.0,
+            pl=1.e5, pr=2.e5,
+            ul=None, ur=None,
+            yl=None, yr=None,
+            uc=None, sigma=0.5
+    ):
+        r"""Initialize mixture parameters.
+
+        Parameters
+        ----------
+        dim: int
+            specifies the number of dimensions for the solution
+        x0: float
+           location of discontinuity
+        nspeces: int
+            specifies the number of mixture species
+        pl: float
+            pressure to the left of the discontinuity
+        tl: float
+            temperature to the left of the discontinuity
+        ul: numpy.ndarray
+            velocity (vector) to the left of the discontinuity
+        yl: numpy.ndarray
+            species mass fractions to the left of the discontinuity
+        pr: float
+            pressure to the right of the discontinuity
+        tr: float
+            temperaure to the right of the discontinuity
+        ur: numpy.ndarray
+            velocity (vector) to the right of the discontinuity
+        yr: numpy.ndarray
+            species mass fractions to the right of the discontinuity
```

Comment:
Seems like this would be a little easier to read if the corresponding `<something>l` and `<something>r` were next to each other?

Reply:
I sort of wonder if they should come in as tuples?

Reply:
I struggle with how to arrange these, it feels natural  to group them by left and right. but grouping by pressure/temperature, etc. makes sense as well.

Reply:
My vote would be to arrange by physical quantity, not by left vs. right.  Either way is perfectly understandable, but I do agree with @majosm's arrangement suggestion.